Add App suffix to assembly name to avoid recursive reference

tpetricek opened the pull request:
I'm not sure how to test this, but if I correctly understand how this works, this PR should fix #27.

tpetricek committed:
Add App suffix to assembly name to avoid recursive reference (fix #27)

TheAngryByrd commented:
To test: 

You should be able to build with `-InstallTemplate` flag set.  Then create a new folder somewhere and try the `dotnet new giraffe` 

(doesn't look like the bash script passes along arguments to the powershell script so you'll have to just invoke the ps1)

dustinmoris reviewed (dismissed):
LGTM

dustinmoris commented:
You can also use the `-TestPermutations` flag to test all the template builds. This will install the new template, then run the dotnet new giraffe command for all possible permutations and also run their individual build scripts which will execute tests if the permuation has tests. I'm currently running this locally...

FYI, this will then create all these different apps in a .temp folder:
<img width="326" alt="image" src="https://user-images.githubusercontent.com/4580714/63635705-6d542500-c65d-11e9-9926-c8fbd2ad489e.png">

dustinmoris committed:
Minor changes

- Updated SDK version in global.json
- Removed version from the Microsoft.AspNetCore.App NuGet package reference in all templates
- Upped the version of the .nuspec
- Added the .ionide folder to the gitignore
